lib/ai/model/tool.go

```diff
@@ -19,16 +19,21 @@ package model
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/gen/proto/go/assist/v1"
+	"github.com/gravitational/teleport/api/types"
 )
 
 // Tool is an interface that allows the agent to interact with the outside world.
 // It is used to implement things such as vector document retrieval and command execution.
 type Tool interface {
 	Name() string
 	Description() string
-	Run(ctx context.Context, input string) (string, error)
+	Run(ctx context.Context, input string) (*stepOutput, error)
```

Comment:
I'm not sure to get all the implications of this change. What happens if the tool returns a stepOutput with `action` set? If the goal is to either return an observation or finish execution, maybe a dedicated structure would be clearer.

Reply:
@xacrimon Do you have any opinions on that? The problem that I have currently is that when a user doesn't have access to any nodes, the empty output confuses the execution, and the LLM keeps asking about the same thing.

Reply:
Preferably can we respond with an english language text hint telling it there are no nodes instead of an empty string? Responding with a stepoutput breaks the nice tool/exec loop encapsulation so not a big fan of this.